src/Umbraco.Web/Compose/UnattendedInstallComponent.cs

```diff
@@ -0,0 +1,46 @@
+﻿using System;
+using Umbraco.Core;
+using Umbraco.Core.Composing;
+using Umbraco.Core.Logging;
+using Umbraco.Core.Migrations.Install;
+using Umbraco.Web.Install;
+
+namespace Umbraco.Web.Compose
```

Comment:
Can we have this in Core and not dependent on any web things? We can avoid the IHttpContextAccessor dependency in some other way? Maybe an event?

Reply:
Yeah, I didn't like putting it in Web. I'll see if it's posible to handle it with events instead.

Reply:
Moved to Core project and added an event that `WebRuntime` subscribes to, it adds `AppPoolRequiresRestart=true` if the installation is successful which is picked up in the `UmbracoInjectedModule` that restarts the application if the runtime is in an install or upgrade state and `AppPoolRequiresRestart=true`.